spring-pulsar-test/src/test/java/org/springframework/pulsar/test/support/ConsumedMessagesConditionTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.apache.pulsar.client.api.Message;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+/**
+ * Tests for {@link ConsumedMessagesCondition}.
+ *
+ * @author Jonas Geiregat
+ */
+class ConsumedMessagesConditionTest {
+
+	@ParameterizedTest
+	@CsvSource({ "true, true, true", "true, false, false", "false, true, false", "false, false, false" })
```

Comment:
I tend to stay away from the `@CsvSource` as it is typically a bit hard on readability of the test. I typically will use the `Arguments` when needed. I think this can be simplified w/ something like: 
```java
	@Test
	void bothConditionsShouldBeMetForAndToBeMet() { 
		var trueCondition = testCondition(true);
		var falseCondition = testCondition(false);
		assertThat(trueCondition.and(trueCondition).meets(List.of())).isTrue();
		assertThat(falseCondition.and(trueCondition).meets(List.of())).isFalse();
		assertThat(trueCondition.and(falseCondition).meets(List.of())).isFalse();
		assertThat(falseCondition.and(falseCondition).meets(List.of())).isFalse();
	}
	
	private ConsumedMessagesCondition<Object> testCondition(boolean b) {
		return messages -> b;
	}

```

Wdyt?

Reply:
With the spring-formatter singelinng the whole statement makes it a bit unreadable.

Reply:
Yep, the formatter does not always do a great job. We try to use this sparingly,  **but** there is the `// @formatter:off|on` that will turn the formatter on/off (e.g.)

```java
	@BeforeAll
	static void startContainers() {
		Network sharedNetwork = Network.newNetwork();
		// @formatter:off
		PULSAR_CONTAINER
				.withNetwork(sharedNetwork)
				.withFunctionsWorker()
				.withClasspathResourceMapping("/connectors/", "/pulsar/connectors", BindMode.READ_ONLY)
				.start();
		RABBITMQ_CONTAINER
				.withNetwork(sharedNetwork)
				.withNetworkAliases("rabbitmq")
				.withExposedPorts(5672, 15672)
				.withStartupTimeout(Duration.ofMinutes(1))
				.start();
		// @formatter:on
	}
```

In this case, I still think a tuple of 3 booleans may be hard to follow (even w/ proper grouping/formatting etc) but I will be happy to take a look if you go that route. 👍🏻